### PR TITLE
Find with selected values when reveal TerminalFind

### LINKED ISSUE #189343: Integrated console find feature uses the previous search
Type: <b>Bug</b>

Steps:
1. Open VS Code
2. Select text in the terminal.
3. Press Ctrl+F to search. You won't see results.
4. Select a different word and Ctrl+F. You will see matches for the previous word searched.
5. Repeat steps 2-4. Same issue.
6. Type a word in search box. It will work.
7. Delete the word in search and repeat steps 2-4. Issue still persists.

https://github.com/microsoft/vscode/assets/65320293/98853d7b-db0d-42ac-993b-aa4ebd5e7a06

https://github.com/microsoft/vscode/assets/65320293/1cf32591-d89c-4d30-bb12-64567f26f39c

VS Code version: Code 1.80.2 (2ccd690cbff1569e4a83d7c43d45101f817401dc, 2023-07-27T20:40:28.909Z)
OS version: Windows_NT x64 10.0.19045
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-7300U CPU @ 2.60GHz (4 x 2712)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|undefined|
|Memory (System)|15.81GB (5.58GB free)|
|Process Argv|--crash-reporter-id 292e6dcb-8991-4c16-97cd-94a326b345e4|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (50)</summary>

Extension|Author (truncated)|Version
---|---|---
pascal|ale|9.6.0
pascal-formatter|ale|2.8.1
vscode-intelephense-client|bme|1.9.5
doxdocgen|csc|1.4.0
githistory|don|0.6.20
go|gol|0.39.1
minifier|Hal|0.0.4
better-cpp-syntax|jef|1.17.2
tws|jki|1.0.1
restructuredtext|lex|189.3.0
add-to-gitignore|mac|1.0.0
Lisp|mat|0.1.12
pretty-formatter|mbl|0.2.2
vscode-docker|ms-|1.26.0
csharp|ms-|1.26.0
vscode-edge-devtools|ms-|2.1.3
black-formatter|ms-|2023.4.1
isort|ms-|2023.10.1
python|ms-|2023.12.0
vscode-pylance|ms-|2023.7.40
jupyter|ms-|2023.6.1101941928
jupyter-keymap|ms-|1.1.2
jupyter-renderers|ms-|1.0.17
vscode-jupyter-cell-tags|ms-|0.1.8
vscode-jupyter-slideshow|ms-|0.1.5
remote-containers|ms-|0.299.0
remote-ssh|ms-|0.102.0
remote-ssh-edit|ms-|0.86.0
remote-wsl|ms-|0.80.2
cmake-tools|ms-|1.14.34
cpptools|ms-|1.16.3
cpptools-extension-pack|ms-|1.3.0
hexeditor|ms-|1.9.12
live-server|ms-|0.4.9
powershell|ms-|2023.6.0
vscode-serial-monitor|ms-|0.10.0
debugger-for-edge|msj|1.0.15
vscode-groovy-lint|Nic|2.0.0
java|red|1.21.0
code-spell-checker|str|2.20.5
simple-rst|tro|1.5.3
cmake|twx|0.0.17
intellicode-api-usage-examples|Vis|0.2.7
vscodeintellicode|Vis|1.2.30
vscode-arduino|vsc|0.6.0
vscode-java-debug|vsc|0.52.0
vscode-java-dependency|vsc|0.23.0
vscode-java-pack|vsc|0.25.12
vscode-java-test|vsc|0.39.0
vscode-maven|vsc|0.41.0

(1 theme extensions excluded)

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
vslsvsres303:30308271
vserr242cf:30382550
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
vscod805cf:30301675
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593:30376534
pythonvs932:30410667
py29gd2263cf:30792227
vsclangdf:30486550
c4g48928:30535728
dsvsc012:30540252
pynewext54:30695312
azure-dev_surveyone:30548225
282f8724:30602487
89544117:30613380
showlangstatbar:30737416
vsctsb:30748421
03d35959:30757346
pythonfmttext:30731395
pythoncmvfstrcf:30756944
fixshowwlkth:30771522
showindicator:30785052
pythongtdpath:30769146
i26e3531:30792625
pythonnosmt12:30797651
pythonnoceb:30797650
e537b577:30795824
dsvsc013:30795093
dsvsc014cf:30797590

```

</details>

### PR DESCRIPTION
**Pull request by tisilent**
fix #189343

### CODE REVIEW

**Review by meganrogge — DISMISSED** [review]

File: src/vs/workbench/contrib/terminalContrib/find/browser/terminalFindWidget.ts [hunk 1]
```diff
@@ -81,9 +81,9 @@ export class TerminalFindWidget extends SimpleFindWidget {
 	override reveal(): void {
 		const initialInput = this._instance.hasSelection() && !this._instance.selection!.includes('\n') ? this._instance.selection : undefined;
 		const xterm = this._instance.xterm;
-		if (xterm && this.inputValue && this.inputValue !== '') {
+		if (xterm && initialInput && initialInput !== '') {
```
Comment (meganrogge) [inline src/vs/workbench/contrib/terminalContrib/find/browser/terminalFindWidget.ts:84]: do we want to fallback to `this.inputValue` if `initialInput` is undefined?

**Review by connor4312 — DISMISSED** [review]

File: src/vs/workbench/contrib/terminalContrib/find/browser/terminalFindWidget.ts (hunk 1 above)
Comment (tisilent) [reply src/vs/workbench/contrib/terminalContrib/find/browser/terminalFindWidget.ts:84]: I need to adjust and test it.

**Comment by tisilent** [comment]
hi @meganrogge ,  fallback to `this.inputValue`, It can improve the following operating experience.
1. reveal
2. hide
3. input command
4. reveal

**Review by Tyriar — APPROVED** [review]
Works great, thanks again @tisilent 👏

**Review by TylerLeonhardt — APPROVED** [review]

**Pull request merged**